app/controllers/admin/payout_reports_controller.rb

```diff
@@ -38,13 +40,21 @@ def upload_settlement_report
     Eyeshade::Publishers.new.create_settlement(body: json)
 
     json.each do |entry|
-      next unless entry["documentId"]
-      invoice = Invoice.find_by(id: entry["documentId"])
+      next unless entry["type"] == MANUAL && entry["owner"] && entry["amount"]
+
+      partner_id = entry["owner"].sub("publishers#uuid:", "")
+
+      invoice = Invoice.where(
+        partner_id: partner_id,
+        finalized_amount: entry["amount"],
```

Comment:
Is there a more elegant way to look this up besides the amount? What happens if there are duplicates? (I realize this is unlikely, but it might bite us in the ass a year from now)

Reply:
Unfortunately not, I think this will be fine because we're basing it on the specific partner, filtering it down to finalized invoices,  and the amount will be the exact same. I think the chaces of duplicates are slim